tensorflow/core/kernels/dml_softmax_ops.cc

```diff
@@ -95,14 +118,43 @@ class DmlSoftmaxXentWithLogitsKernel : public DmlKernel {
     // The strides we need to set to broadcast class across an entire tensor
     dml::TensorDesc::Dimensions reduce_strides = {0, 1, 3};
 
+    // broadcast logits if the shape do not match
+    uint32_t toBCastBatchDim =
+        logits_shape.dim_size(0) == shape_in.dim_size(0) ? 0 : 1;
+    uint32_t toBCastClassDim =
+        logits_shape.dim_size(1) == shape_in.dim_size(1) ? 0 : 1;
+    dml::Expression logits_bcast;
+    if (toBCastBatchDim || toBCastClassDim) {
+      logits_bcast = dml::Reinterpret(
+          logits, input_sizes,
+          dml::TensorDesc::Dimensions(
+              {0, 0, 1 - toBCastBatchDim, 1 - toBCastClassDim}));
+    } else {
+      logits_bcast = dml::Reinterpret(logits, input_sizes, {});
+    }

```

Comment:
The dml::Reintrpret in the `else` clause is basically a no-op. You can simplify your logic by simply removing the `logits_bcast` variable and overwriting `logits` instead. This allows you to remove the `else` clause and gets rid of the uninitialized `logits_bcast` variable.

```suggestion
    if (toBCastBatchDim || toBCastClassDim) {
      logits = dml::Reinterpret(
          logits, input_sizes,
          dml::TensorDesc::Dimensions(
              {0, 0, 1 - toBCastBatchDim, 1 - toBCastClassDim}));
    }
```